src/mod/l4_thruster.c

```diff
@@ -0,0 +1,539 @@
+		//length of payload is address(1)+data(2) = 3 bytes
+		request[4] =3;
+
+	}
+
+	if (length_of_register ==4){
+			/// 4 bytes only for FUSE REGISTERS
+			request[4] =5;
+
+		}
+
+	//For payload length uint16_t is used. However we represent uint16_t as two uint8_t bytes
+	//IMPORTANT NOTE : For request to set a SINGLE register - payload length would obviously
+	// never exceed 255.  Therefore second byte request[5] representing second part of uint16_t
+	// would always be 0x00
+	request[5] = 0x00;
+	request[6]= access_register; // address of register that intended to be set
+
+
+	// Parse arguments and input transformation using conversion multipliers array
+	if (length_of_register ==1){
+		uint8_t input_uint8 = atoi(argv[2]);
+
+		//transform input using conversion multipliers array.
+		// at all registers with only 1 byte data length conversion multiplier array
+		// will also be an integer value. Therefore typecast of conversion multiplier
+		//into uint8_t should also work
+		input_uint8 = input_uint8* (uint8_t) CONVERSION_DOUBLE[access_register];
+		request[7]= input_uint8;
+
+	}
+
+	if (length_of_register ==2){
+
+		// parse argument as double
+		// WARNING : Set Project-Settings-Manager linker script - Redlib (nohost) to use atof()
+		double input = atof((const char*)argv[2]);
+		// in cases where data stored in register has length of 2 bytes. uint16_t would be used
+		// to store this data in register. For some registers input may be float
+		// example 3.3V  or 3.0 or 3
+		// Function should be able to handle both float and integer inputs from user
+
+		// Apply conversion multiplier
+		input = input* CONVERSION_DOUBLE[access_register];
+		// Convert input to uint16_t
+		uint16_t value = (uint16_t) input;
+		// Represent uint16_t as two uint8_t bytes to fill the request array.
+		request[7]= value & 0xff;
+		request[8] = (value >> 8) & 0xff;
+	}
+
+	request[3] = CRC8(request,len); // calculate checksum after whole request array is sent
+	l4_thr_ExpectedReceiveBuffer = 6;// change expected receive buffer accordingly
+	thrSendBytes(request, len);
+}
+
+
+//General read request to any register
+void GeneralReadRequest(int argc, char *argv[]){
+
+		// FIRST ARGUMENT SHOUD BE int VALUE OF REGISTER THAT WOULD BE READ FROM
+		uint8_t access_register = atoi(argv[1]);
+		uint8_t length_of_register = REGISTER_LENGTH[access_register];
+
+		// TODO : Check if valid/existing register is accessed
+		if(length_of_register ==0){
+			return;
+		}
+
+		uint8_t request[8];
+		request[0] = SENDER_ADRESS;
+		request[1] = DEVICE;
+		request[2] = MSGTYPE[2]; // READ -3
+		request[3] = 0x00; // checksum
+		request[4] = 0x02; // LENGTH of payload REGISTER and length
+		request[5] = 0x00; // Hardcoded because length of payload is defined with two bytes of uint16
+		request[6]= access_register; // Access register at address requested by used
+		request[7]= length_of_register;
+
+		uint8_t len = sizeof(request);
+		request[3] = CRC8(request,len);
```

Comment:
not sure the CRC8() coded in ado_crc is exactly th8is polinom (you copied over from hardwaretest repos). If yes pls. use ado function. If not we should make the CRC8 implementation in ado fitter .....